src/interp/mod.rs

```diff
@@ -57,11 +61,14 @@ impl LLVMIRInterpreter {
         }
     }
 
-    fn it_bb(&mut self, func: &Function, bb_name: name::Name) {
-        let mut bb_name = Some(bb_name);
-        while bb_name != None {
-            //PERF: get_bb_by_name function is inefficient.
-            let bb = func.get_bb_by_name(bb_name.as_ref().unwrap()).unwrap();
+    fn it_bb(&mut self, func: &Function) -> Option<Operand> {
+        // for par in &func.parameters {
+        //     println!("cray cray")
+        // }
+
+        let mut bb_option = Some(&func.basic_blocks[0]);
+        while bb_option != None {
```

Comment:
`!bb_option.is_none()`

Reply:
Ah you are unwrapping immediately after. In that case we can do `while let Some(bb) = bb_option`.